### PR TITLE
[1.13.x] [RHPAM- 4269] - RHPAM Kogito Operator 7.12.0-2 uses

### PR DESCRIPTION
**Pull request by spolti**
registry.stage.redhat.io/rhpam-7/rhpam-kogito-runtime-jvm-rhel8:7.12

Incorporates [KOGITO-6949] - Automatic control the csv version on
rhpam-kogito-operator

Signed-off-by: spolti <fspolti@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-operator/blob/main/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains a link to the JIRA issue
- [x] Pull Request contains a description of the issue
- [] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [ ] You've ran `make before-pr` and everything is working accordingly
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster
- [ ] You've added a [RELEASE_NOTES.md](https://github.com/kiegroup/kogito-operator/blob/main/RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run operator BDD testing</b>
  Please add comment: <b>/jenkins test</b>

* <b>Run RHPAM operator BDD testing</b>
  Please add comment: <b>/jenkins RHPAM test</b>
</details>

### CODE REVIEW

File: config/manifests/rhpam/bases/rhpam-kogito-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ metadata:
     capabilities: Basic Install
     categories: Integration & Delivery
     certified: "true"
-    containerImage: registry.stage.redhat.io/rhpam-7/rhpam-kogito-rhel8-operator:7.11.0
+    containerImage: registry.stage.redhat.io/rhpam-7/rhpam-kogito-rhel8-operator:0.0.0
```
[the remark this answers is not in the document]
Comment (spolti) [reply config/manifests/rhpam/bases/rhpam-kogito-operator.clusterserviceversion.yaml:9]: imho it is better not hardcod versions here an d make the version/rhpam/version.go the point of truth, wdyt?

File: version/rhpam/version.go [hunk 2]
```diff
@@ -15,6 +15,12 @@
...
 var (
-	// Version of RHPAM Kogito Operator
+	// PriorVersion product version
+	PriorVersion = "7.12.0"
+	// Version - current version
 	Version = "7.13.0"
+	// CsvVersion - csv release
+	CsvVersion = Version + "-1"
+	// CsvPriorVersion - prior csv release
+	CsvPriorVersion = PriorVersion + "-2"
```
[the remark this answers is not in the document]
Comment (spolti) [reply version/rhpam/version.go:25]: fixed

File: hack/rhpam/rhpam-update-bundle.sh [hunk 3]
```diff
@@ -16,6 +16,24 @@
 
 set -e
 
+script_dir_path=`dirname "${BASH_SOURCE[0]}"`
+source ${script_dir_path}/rhpam-env.sh
+source ${script_dir_path}/rhpam-env.sh
```
[the remark this answers is not in the document]
Comment (spolti) [reply hack/rhpam/rhpam-update-bundle.sh:21]: what exactly you have changed here? :)

File: config/manifests/rhpam/bases/rhpam-kogito-operator.clusterserviceversion.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (radtriste) [reply config/manifests/rhpam/bases/rhpam-kogito-operator.clusterserviceversion.yaml:9]: ok

File: hack/rhpam/rhpam-update-bundle.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (radtriste) [reply hack/rhpam/rhpam-update-bundle.sh:21]: well I removed one line because it was doing it twice ;)

[the remark this answers is not in the document]
Comment (spolti) [reply hack/rhpam/rhpam-update-bundle.sh:21]: LOL thanks.

File: rhpam-operator.yaml [hunk 4]
```diff
@@ -2474,7 +2474,7 @@ spec:
         - name: RUNTIME_IMAGE
           value: rhpam-kogito-runtime-jvm-rhel8
         - name: IMAGE_REGISTRY
-          value: registry.stage.redhat.io/rhpam-7
+          value: registry.redhat.io/rhpam-7
```
Comment (spolti) [inline rhpam-operator.yaml:2477]: @radtriste @AntStephenson @rgdoliveira @jakubschwan 
do you guys think we can use the official registry directly or, is it mandatory to use the stage here?

Comment (spolti) [reply rhpam-operator.yaml:2477]: we can replace it with scripts with the make bundle command.
We tried to rely on osbs but it didn't work as expected.

Comment (spolti) [reply rhpam-operator.yaml:2477]: discussed on gchat.

**Review by desmax74 — APPROVED** [review]

**Review by davidesalerno — APPROVED** [review]

**Review by AntStephenson — APPROVED** [review]

**Pull request merged**